lib/segregate.rb

```diff
@@ -269,4 +276,14 @@ def parse_chunked_data_chunk line
       @stashed_body = @inital_line.end_with?("\r\n") ? (line + "\r\n") : line
     end
   end
+
+  def save_value(store_hash, key, value)
+    key = key.downcase
+    if store_hash.key?(key)
+      store_hash[key] = [] << store_hash[key] unless store_hash[key].kind_of?(Array)
```

Comment:
I did in  this way to maintain retrocompatibility. If I do like you are suggesting I will have to change all the lines where are being done validations like for example:

headers['transfer-encoding'] == 'chunked'